options/glibc/generic/error.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <stdarg.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <error.h>
+
+unsigned int error_message_count = 0;
+int error_one_per_line = 0;
+void (*error_print_progname)(void) = NULL;
+
+void error(int status, int errnum, const char *format, ...) {
+	va_list args;
+	va_list copied;
+	va_start(args, format);
+	va_copy(copied, args);
+	va_end(args);
```

Comment:
I use va_* APIs maybe once a year at this point, so not entirely sure if I'm right, but: isn't the copy here redundant?

Reply:
As far as I know a copy is only needed if you want to iterate through the same varargs twice (e.g. when calling 2 functions that take a `va_list` with the same list).

Reply:
I think I remember something about needing to copy the list of reusing it, but I'm not 100% sure and this API isn't really my strong suit. If this isn't required, would I simply be able to initialize the list and pass it on to `vfprintf`?

Reply:
Yeah if you want to use the same list twice you need to make a copy, but in this case you only use it to pass it on to `vfprintf`, in which case you can just pass the original one through.

Reply:
Makes sense; I'll change this later.

Reply:
This has been corrected.